tests/test_soapclient.py

```diff
@@ -0,0 +1,102 @@
+        body = b''  # body content relevant in this test
+        self._setup_mock(body)
+
+        # normal call, no exception
+        self.soap_client.get_from_url('foo', msg='')
+
+        # http connection raises an exception => exception is delegated to caller
+        for exception in (HTTPException('mock'), OSError('mock'), Exception('mock')):
+            mocked_connection, http_response = self._setup_mock(body)
+
+            mocked_connection.getresponse.side_effect = exception
+            self.assertRaises(exception.__class__, self.soap_client.get_from_url, 'foo', '')
+            self.assertFalse(self.soap_client.is_closed())
+
+    def test_post_message_to(self):
+        """Check error handling of post_message_to method."""
+        # create a message to be sent
+        factory = MessageFactory(SDC_v1_Definitions, None, self.logger)
+        inf = HeaderInformationBlock(action='some_action', addr_to='does_not_matter')
+        payload = Renew()
+        created_message = factory.mk_soap_message(inf, payload)
+
+        # create a message to be received
+        factory = MessageFactory(SDC_v1_Definitions, None, get_logger_adapter('mock'))
+        payload = RenewResponse()
+        payload.Expires = 10
+        inf = HeaderInformationBlock(action='some_action', addr_to='does_not_matter')
+        returned_message = factory.mk_soap_message(inf, payload)
+        body = returned_message.serialize()
+
+        # mock the connection
+        self._setup_mock(body)
+
+        # normal call, no exception
+        result = self.soap_client.post_message_to('renew', created_message, validate=False)
+        self.assertIsInstance(result, ReceivedMessage)
+        self.assertFalse(self.soap_client.is_closed())
+
+        # request method of http connection raises an Exception => exception is converted to NotConnected
+        for exception in (HTTPException('mock'), OSError('mock'), Exception('mock')):
+            mocked_connection, http_response = self._setup_mock(body)
+            mocked_connection.request.side_effect = exception
+            self.assertRaises(NotConnected, self.soap_client.post_message_to, 'renew', created_message, validate=False)
+            self.assertTrue(self.soap_client.is_closed())
+
+        # getresponse method of http connection raises an Exception => exception is converted to NotConnected
+        for exception in (HTTPException('mock'), OSError('mock'), Exception('mock')):
+            mocked_connection, http_response = self._setup_mock(body)
+            mocked_connection.getresponse.side_effect = exception
+            self.assertRaises(NotConnected, self.soap_client.post_message_to, 'renew', created_message, validate=False)
+            self.assertTrue(self.soap_client.is_closed())
+
+        # returned status >= 300 => raise HTTPReturnCodeError, connection stays open
+        mocked_connection, http_response = self._setup_mock(body)
+        http_response.status = 333

```

Comment:
if `>=300` should be tested, use `300` instead of `333`

Reply:
it really does not matter!

Reply:
it does matter. If someone for some reason increases `300` to `>=332` this error would not be detected